### PR TITLE
[VSEE-1008] Add validate scan format docs

### PR DESCRIPTION
**Pull request by rxl7906**
Summary:
* Add docs on how to validate scan report results specify for format/version schema with example.

# Which other branches do you want a technical writer to cherry-pick this PR to (if any)?
1.6 
It's best to PR to the most recent relevant branch and leave all the cherry-picking to the
writer, except where earlier branches require factual changes to the content.
For more information about the branches, see https://github.com/pivotal/docs-tap#branches

### CODE REVIEW

**Comment by rxl7906** [comment]
@ryan-a-baker Can you review this?

Following up on the conversation last week about `sbom-utility` and `cyclonedx-cli` hasn't been working properly with trivy scan report outputs https://vmware.slack.com/archives/G01KG511WS0/p1687372400283349, 

How should I phrase a disclaimer saying that the "sbom-utility" tool may not work on the scan report because it is up to the scan vendor to properly output the scan report?

I was thinking along the lines:
```
**Note**: This sbom-utility tool may or may not work with all scan report outputs depending on the scan vendor.
```

I confirmed the sbom-utility works with grype scan report output as shown in the example however with the trivy scan report output I did get a validation error:
```
lrobin@lrobin6MD6R sbom-utility-v0.11.0-darwin-amd64 % ./sbom-utility validate -i ~/go/src/gitlab/app-scanning/scan-results-trivy-json/scan.json
Welcome to the sbom-utility! Version `v0.11.0` (sbom-utility) (darwin/amd64)
============================================================================
[INFO] Loading license policy config file: `license.json`...
[WARN] Invalid flag for command: `output-file` (`o`). Ignoring...
[INFO] Attempting to load and unmarshal file `/Users/lrobin/go/src/gitlab/app-scanning/scan-results-trivy-json/scan.json`...
[INFO] Successfully unmarshalled data from: `/Users/lrobin/go/src/gitlab/app-scanning/scan-results-trivy-json/scan.json`
[INFO] Determining file's SBOM format and version...
[INFO] Determined SBOM format, version (variant): `CycloneDX`, `1.4` (latest)
[INFO] Matching SBOM schema (for validation): schema/cyclonedx/1.4/bom-1.4.schema.json
[INFO] Loading schema `schema/cyclonedx/1.4/bom-1.4.schema.json`...
[INFO] Schema `schema/cyclonedx/1.4/bom-1.4.schema.json` loaded.
[INFO] Validating `/Users/lrobin/go/src/gitlab/app-scanning/scan-results-trivy-json/scan.json`...
[INFO] SBOM valid against JSON schema: `false`
[ERROR] invalid SBOM: schema errors found (/Users/lrobin/go/src/gitlab/app-scanning/scan-results-trivy-json/scan.json):
(2) Schema errors detected (use `--debug` for more details):
        1. Type: [unique], Field: [dependencies.37.dependsOn], Description: [array items[0,1] must be unique]
        Failing object: [[
          "pkg:deb/debian/libc6@2.31-13+deb11u5?arch=amd64&distro=debian-11.6",
          "pkg:deb/debi ... (truncated)
        2. Type: [unique], Field: [dependencies.79.dependsOn], Description: [array items[0,1] must be unique]
        Failing object: [[
          "pkg:deb/debian/debconf@1.5.77?arch=all&distro=debian-11.6",
          "pkg:deb/debian/debcon ... (truncated)
[INFO] document `/Users/lrobin/go/src/gitlab/app-scanning/scan-results-trivy-json/scan.json`: valid=[false]

```

File: scst-scan/verify-app-scanning.hbs.md [hunk 1]
```diff
@@ -51,3 +51,31 @@ To retrieve a vulnerability report:
    imgpkg pull -b $SCAN_RESULT_URL -o scan-results/
    ls scan-results/
    ```
+
+## <a id="validating-scan-format"></a> Validating Scan Format
+
+After retrieving the scan results, the scan results can be validated via this CycloneDX tool, [sbom-utility](https://github.com/CycloneDX/sbom-utility). This tool is designed to validate CycloneDX and SPDX BOMs against versioned schemas.
+1. Setup and install using the instructions [here](https://github.com/CycloneDX/sbom-utility#installation).
+2. Run the `sbom-utility` CLI with the subcommand [validate](https://github.com/CycloneDX/sbom-utility#validate) to validate the scan report against its declared format (e.g., SPDX, CycloneDX) and version (e.g., "2.2", "1.4", etc.).
```
Comment (ryan-a-baker) [inline scst-scan/verify-app-scanning.hbs.md:59]: Can we put a table here that specifies supported version?  SPDX 2.2 and Cyclone DX 1.2/1.3./1.4?  We should tell the user here what formats to validate against that AMR Observer supports and not leave them guessing.

**Comment by ryan-a-baker** [comment]
```
**Note**: This sbom-utility tool may or may not work with all scan report outputs depending on the scan vendor.
```

I read this as there is something wrong with the sbom-utility, when in actuality it is the output of the scanner that is not in the correct format that is the problem.  @bsoroushian made the comment that just because it's not valid doesn't mean that we can't parse it, which is true.  However, we _should not_ certify/support the scanner as an integration if the output is not valid per SPDX/CycloneDX format as there is no way that we can scale to support a "Bring Your Own Scanner" model if the scanners are not output in valid format.

I think the disclaimer should be that the output of the scan should be valid per SPDX/CycloneDX spec, and if not, although it may be parsed correctly, VMware can not ensure that the information will be parsed correctly, and results may not be displayed accurately in TAP GUI and TAP CLI.

File: scst-scan/verify-app-scanning.hbs.md [hunk 2]
```diff
@@ -51,3 +51,31 @@ To retrieve a vulnerability report:
    imgpkg pull -b $SCAN_RESULT_URL -o scan-results/
    ls scan-results/
    ```
+
+## <a id="validating-scan-format"></a> Validating Scan Format
+
+After retrieving the scan results, the scan results can be validated via this CycloneDX tool, [sbom-utility](https://github.com/CycloneDX/sbom-utility). This tool is designed to validate CycloneDX and SPDX BOMs against versioned schemas.
```
Comment (ryan-a-baker) [inline scst-scan/verify-app-scanning.hbs.md:57]:
```suggestion
After retrieving the scan results, the scan results must be validated to be in a format that downstream Tanzu Application Platform services such as AMR observer support.  The recommended way to validate the scan results is via this CycloneDX tool, [sbom-utility](https://github.com/CycloneDX/sbom-utility). This tool is designed to validate CycloneDX and SPDX BOMs against versioned schemas.
```

File: scst-scan/verify-app-scanning.hbs.md (hunk 1 above)
Comment (ryan-a-baker) [reply scst-scan/verify-app-scanning.hbs.md:59]: Probably makes sense for the table to be here:

https://github.com/pivotal/docs-tap/pull/2647/files#r1242626833

Comment (rxl7906) [reply scst-scan/verify-app-scanning.hbs.md:59]: I could put a table however the `sbom-utility` tool itself does provide a CLI command to show what schemas/version/format are available (https://github.com/CycloneDX/sbom-utility#schema). 

```
lrobin@lrobin6MD6R sbom-utility-v0.11.0-darwin-amd64 % ./sbom-utility schema list
Welcome to the sbom-utility! Version `v0.11.0` (sbom-utility) (darwin/amd64)
============================================================================
[INFO] Loading license policy config file: `license.json`...
[INFO] Scanning document for vulnerabilities...
name                          format     version   variant      file                                             url
----                          ------     -------   -------      ----                                             ---
CycloneDX v1.5 (development)  CycloneDX  1.5       development  schema/cyclonedx/1.5/bom-1.5-dev.schema.json     https://raw.githubusercontent.com/CycloneDX/specification/v1.5-dev/schema/bom-1.5.schema.json
CycloneDX v1.4                CycloneDX  1.4       (latest)     schema/cyclonedx/1.4/bom-1.4.schema.json         https://raw.githubusercontent.com/CycloneDX/specification/master/schema/bom-1.4.schema.json
CycloneDX v1.4 (custom)       CycloneDX  1.4       custom       schema/test/bom-1.4-custom.schema.json
CycloneDX v1.3                CycloneDX  1.3       (latest)     schema/cyclonedx/1.3/bom-1.3.schema.json         https://raw.githubusercontent.com/CycloneDX/specification/master/schema/bom-1.3.schema.json
CycloneDX v1.3 (custom)       CycloneDX  1.3       custom       schema/test/bom-1.3-custom.schema.json
CycloneDX v1.3 (strict)       CycloneDX  1.3       strict       schema/cyclonedx/1.3/bom-1.3-strict.schema.json  https://raw.githubusercontent.com/CycloneDX/specification/master/schema/bom-1.3-strict.schema.json
CycloneDX v1.2                CycloneDX  1.2       (latest)     schema/cyclonedx/1.2/bom-1.2.schema.json         https://raw.githubusercontent.com/CycloneDX/specification/master/schema/bom-1.2.schema.json
CycloneDX v1.2 (strict)       CycloneDX  1.2       strict       schema/cyclonedx/1.2/bom-1.2-strict.schema.json  https://raw.githubusercontent.com/CycloneDX/specification/master/schema/bom-1.2-strict.schema.json
SPDX v2.3                     SPDX       SPDX-2.3  (latest)     schema/spdx/2.3/spdx-schema.json                 https://raw.githubusercontent.com/spdx/spdx-spec/development/v2.3/schemas/spdx-schema.json
SPDX v2.3.1 (development)     SPDX       SPDX-2.3  development  schema/spdx/2.3.1/spdx-schema.json               https://raw.githubusercontent.com/spdx/spdx-spec/development/v2.3.1/schemas/spdx-schema.json
SPDX v2.2.2                   SPDX       SPDX-2.2  (latest)     schema/spdx/2.2.2/spdx-schema.json               https://raw.githubusercontent.com/spdx/spdx-spec/v2.2.2/schemas/spdx-schema.json
SPDX v2.2.1                   SPDX       SPDX-2.2  2.2.1        schema/spdx/2.2.1/spdx-schema.json               https://raw.githubusercontent.com/spdx/spdx-spec/v2.2.1/schemas/spdx-schema.json
```

There's also a way to add more schemas manually with a config file https://github.com/CycloneDX/sbom-utility#adding-sbom-formats-schema-versions-and-variants. 

I think we can tell the user they can run the `./sbom-utility schema list` command to show the supported versions/format/schemas. What do you think?

**Comment by rxl7906** [comment]
@ryan-a-baker can you review again? I made your requested changes except I put the formatting "table" in step 2 because it was lengthy but let me know if you still think it should be placed in the main summary of the section.

**Review by ryan-a-baker — APPROVED** [review]

**Pull request merged**